pkg/adapter/awssnssource/adapter.go

```diff
@@ -18,216 +18,155 @@ package awssnssource
 		ceClient cloudevents.Client) pkgadapter.Adapter {
 
-		logger := logging.FromContext(ctx)
-
-		env := envAcc.(*envConfig)
-
-		arn := common.MustParseARN(env.ARN)
-
-		cfg := session.Must(session.NewSession(aws.NewConfig().
-			WithRegion(arn.Region).
-			WithMaxRetries(5),
-		))
-
 		ns := injection.GetNamespaceScope(ctx)
+		secrGetter := secretGetter(k8sclient.Get(ctx).CoreV1().Secrets(ns))
+		srcClient := client.Get(ctx).SourcesV1alpha1().AWSSNSSources(ns)
 
 		return &adapter{
-			logger: logger,
+			logger: logging.FromContext(ctx),
 
-			snsClient: sns.New(cfg),
-			ceClient:  ceClient,
+			ceClient: ceClient,
+			snsCg:    snsclient.NewClientGetter(secrGetter),
 
-			arn: arn,
-
-			statusPatcher: status.NewPatcher(component, ns, ctx),
+			router:        &router.Router{},
+			statusPatcher: status.NewPatcher(component, srcClient),
 		}
 	}
 }
 
+func secretGetter(cli coreclientv1.SecretInterface) snsclient.NamespacedSecretsGetter {
+	return func(string) coreclientv1.SecretInterface {
+		return cli
+	}
+}
+
 const (
-	serverPort                = "8080"
-	serverShutdownGracePeriod = time.Second * 10
+	serverPort                uint16 = 8080
+	serverShutdownGracePeriod        = time.Second * 10
 )
 
 // Start implements adapter.Adapter.
 func (a *adapter) Start(ctx context.Context) error {
-	http.HandleFunc("/", a.handleNotification)
-	http.HandleFunc("/health", healthCheckHandler)
+	server := &http.Server{
+		Addr:    fmt.Sprint(":", serverPort),
+		Handler: a,
+	}
+
+	return runHandler(ctx, server)
+}
 
-	server := &http.Server{Addr: ":" + serverPort}
-	serverErrCh := make(chan error)
-	defer close(serverErrCh)
+// runHandler runs the HTTP event handler until ctx get cancelled.
+func runHandler(ctx context.Context, s *http.Server) error {
+	logging.FromContext(ctx).Info("Starting HTTP event handler")
 
+	errCh := make(chan error)
 	go func() {
-		a.logger.Info("Serving on port " + serverPort)
-		serverErrCh <- server.ListenAndServe()
+		errCh <- s.ListenAndServe()
 	}()
 
-	var err error
-
-	select {
-	case serverErr := <-serverErrCh:
-		if serverErr != nil {
-			err = fmt.Errorf("failure during runtime of SNS notification handler: %w", serverErr)
+	handleServerError := func(err error) error {
+		if err != http.ErrServerClosed {
+			return fmt.Errorf("during server runtime: %w", err)
 		}
+		return nil
+	}
 
+	select {
 	case <-ctx.Done():
-		a.logger.Info("Shutting server down")
+		logging.FromContext(ctx).Info("HTTP event handler is shutting down")
 
 		ctx, cancel := context.WithTimeout(context.Background(), serverShutdownGracePeriod)
 		defer cancel()
-		if shutdownErr := server.Shutdown(ctx); shutdownErr != nil {
-			err = fmt.Errorf("error during server shutdown: %w", shutdownErr)
-		}
 
-		// unblock server goroutine
-		<-serverErrCh
-	}
-
-	return err
-}
-
-// handleNotification implements the receive interface for SNS.
-func (a *adapter) handleNotification(rw http.ResponseWriter, r *http.Request) {
```

Comment:
This handling logic was moved to `handler/handler.go`. It is now invoked by each handler's ServeHTTP method, and not the adapter directly.